app/controllers/application_controller.rb

```diff
@@ -32,22 +37,38 @@ def default_url_options(options = {})
 
     if current_organization.present? && !options.key?(:organization_id)
       options[:organization_id] = current_organization.to_param
-    elsif current_user && !current_user.super_admin? && current_user.organization.present?
+    elsif current_role.name == "org_admin"
       options[:organization_id] = current_user.organization.to_param
-    elsif current_user&.super_admin?
+    elsif current_role.name == "super_admin"
       # FIXME: This *might* not be the best way to approach this...
       options[:organization_id] = "admin"
     end
     options
   end
 
+  def dashboard_path_from_role
+    if current_role.name == "super_admin"
+      admin_dashboard_path
+    elsif current_role.name == "partner"
+      partners_dashboard_path
+    elsif current_user.organization
+      dashboard_path(current_user.organization)
+    else
+      root_path
+    end
+  end
+
   def authorize_user
     return unless params[:controller] # part of omniauth controller flow
-    verboten! unless params[:controller].include?("devise") || current_user.super_admin? || current_organization.id == current_user.organization_id
+    verboten! unless params[:controller].include?("devise") ||
+      current_user.has_role?(:super_admin) ||
+      current_user.has_role?(:org_user, current_organization) ||
+      current_user.has_role?(:org_admin, current_organization)
```

Comment:
@dorner I added this in but now looking a bit closer to the code and implementation path, I'am wondering if we want to require every `admin` to also be a `user`? Or just one? I see in the code we actually grant them both roles. Aka should we require a user to be a user ontop of being an admin?

Reply:
Yep, we should have both roles. This way we can still get a list of all users just by looking at the `org_user` role, and use the `org_admin` role for additional permissions.